src/sqlancer/doris/DorisSchema.java

```diff
@@ -0,0 +1,418 @@
+                    size = 1;
+                    break;
+                case "DATETIMEV2":
+                    primitiveType = DorisDataType.DATETIME;
+                    size = 2;
+                    break;
+                case "BOOLEAN":
+                    primitiveType = DorisDataType.BOOLEAN;
+                    break;
+                case "HLL":
+                    primitiveType = DorisDataType.HLL;
+                    break;
+                case "BITMAP":
+                    primitiveType = DorisDataType.BITMAP;
+                    break;
+                case "NULL":
+                    primitiveType = DorisDataType.NULL;
+                    break;
+                default:
+                    throw new AssertionError(typeString);
+            }
+        }
+        return new DorisCompositeDataType(primitiveType, size);
+    }
+
+    public static class DorisTable extends AbstractRelationalTable<DorisColumn, TableIndex, DorisGlobalState> {
+
+        public DorisTable(String tableName, List<DorisColumn> columns, boolean isView) {
+            super(tableName, columns, Collections.emptyList(), isView);
+        }
+
+    }
+
+    public static DorisSchema fromConnection(SQLConnection con, String databaseName) throws SQLException {
+        List<DorisTable> databaseTables = new ArrayList<>();
+        List<String> tableNames = getTableNames(con);
+        for (String tableName : tableNames) {
+            if (DBMSCommon.matchesIndexName(tableName)) {
+                continue; // TODO: unexpected?
+            }
+            List<DorisColumn> databaseColumns = getTableColumns(con, tableName);
+            boolean isView = tableName.startsWith("v");
+            DorisTable t = new DorisTable(tableName, databaseColumns, isView);
+            for (DorisColumn c : databaseColumns) {
+                c.setTable(t);
+            }
+            databaseTables.add(t);
+
+        }
+        return new DorisSchema(databaseTables);
+    }
+
+    private static List<String> getTableNames(SQLConnection con) throws SQLException {
+        List<String> tableNames = new ArrayList<>();
+        try (Statement s = con.createStatement()) {
+            try (ResultSet rs = s.executeQuery("SHOW TABLES")) {
+                while (rs.next()) {
+                    tableNames.add(rs.getString(1));
+                }
+            }
+        }
+        return tableNames;
+    }
+
+    private static List<DorisColumn> getTableColumns(SQLConnection con, String tableName) throws SQLException {
+        List<DorisColumn> columns = new ArrayList<>();
+        try (Statement s = con.createStatement()) {
+            try (ResultSet rs = s.executeQuery("DESCRIBE " + tableName)) {
+                while (rs.next()) {
+                    String columnName = rs.getString("Field");
+                    String dataType = rs.getString("Type");
+                    boolean isNullable = rs.getString("Null").contentEquals("Yes");
```

Comment:
Perhaps it would be helpful to add an assert here to check that the value is either "Yes" or (presumably) "No"? Having an assert would help in case this changes in future versions (e.g., to "True").